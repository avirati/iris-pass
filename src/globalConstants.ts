export enum Category {
  BUSINESS        = 'Business',
  CAREER          = 'Career',
  ENTERTAINMENT   = 'Entertainment',
  FINANCE         = 'Finance',
  HEALTH          = 'Health',
  NEWS            = 'News',
  OTHER           = 'Other',
  REAL_ESTATE     = 'Real Estate',
  SHOPPING        = 'Shopping',
  SOCIAL_MEDIA    = 'Social Media',
  TECH            = 'Tech',
  TRAVEL          = 'Travel',
};

export const categories = [
  Category.BUSINESS,
  Category.CAREER,
  Category.ENTERTAINMENT,
  Category.FINANCE,
  Category.HEALTH,
  Category.NEWS,
  Category.OTHER,
  Category.REAL_ESTATE,
  Category.SHOPPING,
  Category.SOCIAL_MEDIA,
  Category.TECH,
  Category.TRAVEL,
];

export const LETTERS_LOWERCASE = 'abcdefghijklmnopqrstuvwxyz'.split('');
export const LETTERS_UPPERCASE = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
export const NUMBERS = '0123456789'.split('');
export const SYMBOLS = '!@#$%^&*()-_+=~[]{};:"\'/<>/?'.split('');