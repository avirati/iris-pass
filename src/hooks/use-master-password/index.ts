export { useMasterPassword } from './useMasterPassword';
export { UseMasterPasswordProvider } from './useMasterPassword.context';
