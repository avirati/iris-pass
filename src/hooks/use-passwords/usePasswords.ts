import { useContext } from 'react'

import { UsePasswordContext } from './usePasswords.context'

export const usePasswords = () => {
  const {
    passwords,
    copyPassword,
    addPassword,
    getPassword,
    removePassword,
    updatePassword,
  } = useContext(UsePasswordContext)

  return {
    passwords,
    copyPassword,
    addPassword,
    getPassword,
    removePassword,
    updatePassword
  }
}