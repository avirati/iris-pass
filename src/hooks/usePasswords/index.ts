export { usePasswords } from './usePasswords'
export { UsePasswordProvider } from './usePasswords.context'
export type { IPassword } from './usePasswords.types'