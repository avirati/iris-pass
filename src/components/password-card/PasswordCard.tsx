import React from 'react'

import { IPassword } from 'hooks/usePasswords'
import { Box, Flex, Heading, Text } from 'shared-components'
import { parseWebsite } from 'utils'

interface IPasswordCard {
  password: IPassword
}

export const PasswordCard: React.FC<IPasswordCard> = ({ password }) => {
  const { website, login } = password
  const { heroLetter, hostname } = parseWebsite(website)

  return (
    <Flex css={{ gap: '$4' }}>
      <Box css={{ p: '$4', border: '1px solid $primary', borderRadius: '$0' }}>
        <Heading
          as="h2"
          size="lg"
          css={{
            textTransform: 'uppercase',
            color: '$primary'
          }}>
            {heroLetter}
        </Heading>
      </Box>
      <Flex css={{ flexDirection: 'column', justifyContent: 'center', gap: '$4' }}>
        <Heading as="h3" size="sm" css={{ color: '$tonal400' }}>{hostname}</Heading>
        <Text size="sm" css={{ color: '$tonal300' }}>{login}</Text>
      </Flex>
    </Flex>
  )
}