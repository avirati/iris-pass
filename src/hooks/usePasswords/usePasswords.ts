import { useContext } from 'react'

import { UsePasswordContext } from './usePasswords.context'

export const usePasswords = () => {
  const {
    passwords,
    addPassword,
    removePassword,
    updatePassword,
  } = useContext(UsePasswordContext)

  return {
    passwords,
    addPassword,
    removePassword,
    updatePassword
  }
}