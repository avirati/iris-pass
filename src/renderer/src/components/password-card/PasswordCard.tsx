import React from 'react';

import { IPassword } from '../../hooks/use-passwords';
import { CSS, Flex, Heading, Link, Text } from '../../shared-components';
import { parseWebsite } from '../../utils';

interface IPasswordCard {
  password: IPassword;
  css?: CSS;
}

const ellipsisCSS: CSS = {
  whiteSpace: 'nowrap',
  overflow: 'hidden',
  textOverflow: 'ellipsis',
};

export const PasswordCard: React.FC<IPasswordCard> = ({ password, css }) => {
  const { website, login, id } = password;
  const { heroLetter, hostname } = parseWebsite(website);

  return (
    <Flex
      as={Link}
      css={{
        gap: '$4',
        '&:before': { display: 'none' },
        '&:hover, &:active, &:focus': {
          color: '$tonal100',
          textDecoration: 'none',
        },
        ...css,
      }}
      href={`#/password/view/${id}`}
    >
      <Flex
        css={{
          minWidth: '$5',
          minHeight: '$5',
          border: '1px solid $tonal200',
          borderRadius: '$0',
          alignItems: 'center',
          justifyContent: 'center',
        }}
      >
        <Heading
          as='h2'
          size='lg'
          css={{
            textTransform: 'uppercase',
            color: '$tonal200',
          }}
        >
          {heroLetter}
        </Heading>
      </Flex>
      <Flex
        css={{
          flexDirection: 'column',
          justifyContent: 'center',
          gap: '$2',
          maxWidth: '250px',
        }}
      >
        <Heading
          as='h3'
          size='md'
          css={{ pb: '$1', color: '$tonal200', ...ellipsisCSS }}
        >
          {hostname}
        </Heading>
        <Text size='sm' css={{ pb: '$1', color: '$tonal200', ...ellipsisCSS }}>
          {login}
        </Text>
      </Flex>
    </Flex>
  );
};
