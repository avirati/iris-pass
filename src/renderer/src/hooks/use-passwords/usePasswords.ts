import { useContext } from 'react';

import { UsePasswordContext } from './usePasswords.context';

export const usePasswords = () => {
  const context = useContext(UsePasswordContext);
  if (!context) {
    throw new Error(
      'usePasswords hook must be used under a UsePasswordContext Provider'
    );
  }

  return { ...context };
};
