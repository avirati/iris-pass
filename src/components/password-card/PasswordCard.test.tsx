import { render } from '@testing-library/react'

import { Category } from 'globalConstants'
import { IPassword } from 'hooks/usePasswords'

import { PasswordCard } from './PasswordCard'

const password: IPassword = {
  id: 'test-id',
  category: Category.BUSINESS,
  login: 'test-login',
  password: 'test-password',
  website: 'https://www.google.com'
}

describe('PasswordCard Component', () => {
  it('renders', () => {
    const { container } = render(<PasswordCard password={password}/>)
    expect(container).toMatchSnapshot()
  })
})