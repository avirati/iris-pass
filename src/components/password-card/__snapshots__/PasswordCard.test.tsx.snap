// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordCard Component renders 1`] = `
<div>
  <div
    class="c-dhzjXW c-dhzjXW-iiSZeJd-css"
  >
    <div
      class="c-PJLV c-PJLV-igGuwjL-css"
    >
      <h2
        class="c-iSnHBU c-iSnHBU-bSbzjX-size-lg c-iSnHBU-ihJOZMv-css"
      >
        w
      </h2>
    </div>
    <div
      class="c-dhzjXW c-dhzjXW-ifKuKYp-css"
    >
      <h3
        class="c-iSnHBU c-iSnHBU-dtbHih-size-sm c-iSnHBU-ildsiYK-css"
      >
        www.google.com
      </h3>
      <p
        class="c-fIXJvv c-fIXJvv-bndJoy-size-sm c-fIXJvv-iesazQZ-css"
      >
        test-login
      </p>
    </div>
  </div>
</div>
`;
